### PR TITLE
Feature/prime 38 self dec upload

### PR DESCRIPTION
**Pull request by hogmann**

### CODE REVIEW

File: prime-dotnet-webapi/ApiDbContext.cs [hunk 1]
```diff
@@ -260,6 +264,21 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                 .WithMany(s => s.BusinessLicences)
                 .HasForeignKey(bl => bl.SiteId);
 
+            modelBuilder.Entity<SelfDeclarationDocument>()
```
Comment (james-hollinger) [inline prime-dotnet-webapi/ApiDbContext.cs:267]: These manual configurations are not needed. EF configures this automatically with Convention Over Configuration (by calling the foreign keys EnrolleId for example)

File: prime-dotnet-webapi/Models/SelfDeclaration.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+        [JsonIgnore]
+        public Enrollee Enrollee { get; set; }
+
+        [Required]
+        public int SelfDeclarationTypeCode { get; set; }
+
+        public SelfDeclarationType SelfDeclarationType { get; set; }
+
+        public string SelfDeclarationDetails { get; set; }
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Models/SelfDeclaration.cs:26]: How about just "Details"?

File: prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs [hunk 3]
```diff
@@ -18,12 +18,9 @@ public class SelfDeclarationRule : AutomaticAdjudicationRule
     {
         public override Task<bool> ProcessRule(Enrollee enrollee)
         {
-            // check to see if any of the self-declaration rules were answered as 'Yes'
+            // check to see if the enrollee has any self declarations
             // note: if for some reason the question was not answered, we will assume 'Yes'
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs:22]: This comment can be removed as well

File: prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs [hunk 4]
```diff
@@ -18,12 +18,9 @@ public class SelfDeclarationRule : AutomaticAdjudicationRule
...
-            if (enrollee.HasConviction.GetValueOrDefault(true)
-                || enrollee.HasDisciplinaryAction.GetValueOrDefault(true)
-                || enrollee.HasPharmaNetSuspended.GetValueOrDefault(true)
-                || enrollee.HasRegistrationSuspended.GetValueOrDefault(true))
+            if (enrollee.SelfDeclarations != null && enrollee.SelfDeclarations.Any())
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs:23]: Im pretty sure SelfDeclarations will be null only if we forget to .Include() the collection. Its probably bad if this rule passes if someone has self-declared and we accidentally remove the include. Null collection should probably also fail the rule.

File: prime-dotnet-webapi/Controllers/EnrolleesController.cs [hunk 5]
```diff
@@ -558,5 +558,32 @@ public async Task<ActionResult> sendEnrolleeReminderEmail(int enrolleeId)
...
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public async Task<ActionResult<SelfDeclarationDocument>> createSelfDeclarationDocument(int enrolleeId, int selfDeclarationTypeCode, SelfDeclarationDocument selfDeclarationDocument)
+        {
+            var enrollee = await _enrolleeService.GetEnrolleeAsync(enrolleeId);
+
+            if (enrollee == null)
+            {
+                return NotFound(ApiResponse.Message($"Enrollee not found with id {enrolleeId}"));
+            }
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/EnrolleesController.cs:582]: need the User.CanAccess check here

File: prime-dotnet-webapi/Models/SelfDeclarationDocument.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        [Required]
+        public int EnrolleeId { get; set; }
+
+        [JsonIgnore]
+        public Enrollee Enrollee { get; set; }
+
+        [Required]
+        public int SelfDeclarationTypeCode { get; set; }
+
+        public SelfDeclarationType SelfDeclarationType { get; set; }
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Models/SelfDeclarationDocument.cs:21]: should probably be [JsonIgnore]

File: prime-dotnet-webapi/Controllers/EnrolleesController.cs [hunk 7]
```diff
@@ -558,5 +558,32 @@ public async Task<ActionResult> sendEnrolleeReminderEmail(int enrolleeId)
 
             return NoContent();
         }
+
+        // POST: api/Enrollees/5/self-declaration-document/3?{GUID}
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/EnrolleesController.cs:562]: This comment seems off

File: prime-dotnet-webapi/Controllers/EnrolleesController.cs [hunk 8]
```diff
@@ -558,5 +558,32 @@ public async Task<ActionResult> sendEnrolleeReminderEmail(int enrolleeId)
...
+        /// <param name="enrolleeId"></param>
+        /// <param name="selfDeclarationTypeCode"></param>
+        /// <param name="selfDeclarationDocument"></param>
+        [HttpPost("{enrolleeId}/self-declaration-document/{selfDeclarationTypeCode}", Name = nameof(createSelfDeclarationDocument))]
+        [ProducesResponseType(typeof(ApiBadRequestResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public async Task<ActionResult<SelfDeclarationDocument>> createSelfDeclarationDocument(int enrolleeId, int selfDeclarationTypeCode, SelfDeclarationDocument selfDeclarationDocument)
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/EnrolleesController.cs:575]: This endpoint should probably accept a type code and a document GUID, and construct the document link in the service. At the very least, we shouldnt use the upload date that the frontend sends.

File: prime-dotnet-webapi/Models/SelfDeclaration.cs [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace Prime.Models
+{
+    [Table("SelfDeclaration")]
+    public class SelfDeclaration : BaseAuditable, IEnrolleeNavigationProperty
+    {
+        [Key]
+        public int Id { get; set; }
+
+        [Required]
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Models/SelfDeclaration.cs:15]: [Required] just means not null, so this doesn't make any difference on a field of type int.

File: prime-angular-frontend/src/app/modules/site-registration/site-registration.module.ts [hunk 10]
```diff
@@ -5,10 +5,10 @@ import { SharedModule } from '@shared/shared.module';
 import { SiteRegistrationRoutingModule } from './site-registration-routing.module';
 
 import { BusinessHoursModule } from '@lib/modules/business-hours/business-hours.module';
-// TODO split out all related filepond files into a /lib module ie. config and components
-import { FilePondModule, registerPlugin } from 'ngx-filepond';
-import FilePondPluginFileValidateType from 'filepond-plugin-file-validate-type';
-registerPlugin(FilePondPluginFileValidateType);
+// // TODO split out all related filepond files into a /lib module ie. config and components
```
Comment (mtpultz) [inline prime-angular-frontend/src/app/modules/site-registration/site-registration.module.ts:8]: Looks like we can remove the comments

**Pull request closed without merging**